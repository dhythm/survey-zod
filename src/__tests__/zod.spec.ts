import { z, ZodError } from "zod";

describe("zod", () => {
  describe("priority", () => {
    const schema = z.object({
      password: z
        .string()
        .min(8)
        .max(16)
        .refine((data) => data.match(/[a-z]/g) && data.match(/[A-Z]/g), {
          message: "password must include capital letter and small letter.",
        })
        .refine((data) => data.match(/[0-9]/g) && data.match(/[a-zA-Z]/g), {
          message: "password must include alphanumeric.",
        })
        .refine(
          (data) =>
            // https://javascript.info/regexp-escaping
            // https://javascript.info/regexp-character-sets-and-ranges#escaping-in-
            data.match(/[!"#\$%&'\(\)\*\+\-\.,\/:;<=>\?@\[\\\]\^_`{\|}~]/g),
          {
            message: "password must include special character.",
          }
        ),
    });
    it("should be passed", () => {
      try {
        schema.parse(
          "12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901"
        );
      } catch (err) {
        expect(err as ZodError).toEqual({});
      }
    });
  });
});
