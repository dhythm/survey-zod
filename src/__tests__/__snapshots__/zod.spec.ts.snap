// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zod schema should be passed 1`] = `
[ZodError: [
  {
    "code": "too_big",
    "maximum": 16,
    "type": "string",
    "inclusive": true,
    "message": "String must contain at most 16 character(s)",
    "path": [
      "password"
    ]
  },
  {
    "code": "custom",
    "message": "password must include capital letter and small letter.",
    "path": [
      "password"
    ]
  },
  {
    "code": "custom",
    "message": "password must include alphanumeric.",
    "path": [
      "password"
    ]
  },
  {
    "code": "custom",
    "message": "password must include special character.",
    "path": [
      "password"
    ]
  }
]]
`;
